core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -166,6 +170,34 @@ private[spark] class ChunkedByteBuffer(var chunks: Array[ByteBuffer]) {
 
 }
 
+object ChunkedByteBuffer {
+  // TODO eliminate this method if we switch BlockManager to getting InputStreams
+  def fromManagedBuffer(data: ManagedBuffer, maxChunkSize: Int): ChunkedByteBuffer = {
+    data match {
+      case f: FileSegmentManagedBuffer =>
+        map(f.getFile, maxChunkSize, f.getOffset, f.getLength)
+      case other =>
+        new ChunkedByteBuffer(other.nioByteBuffer())
+    }
+  }
+
+  def map(file: File, maxChunkSize: Int, offset: Long, length: Long): ChunkedByteBuffer = {
+    Utils.tryWithResource(new FileInputStream(file).getChannel()) { channel =>
```

Comment:
Can we please use `FileChannel#open` instead, FileInputStream/FileOutputStream has some issues (https://www.cloudbees.com/blog/fileinputstream-fileoutputstream-considered-harmful)

Reply:
I wasn't aware of that issue, thanks for sharing that, I'll update this.  Should we also update other uses?  Seems there are a lot of other cases, eg. `UnsafeShuffleWriter`, `DiskBlockObjectWriter`, etc.

Reply:
I've already updated some of them in SPARK-21475 in shuffle related code path, but not all of them which are not so critical.

Reply:
great, thanks for the explanation